docs/proposals/PROPOSAL_C.md

```diff
@@ -0,0 +1,175 @@
+
+1. **Simple checking of key/value pairs**: A community might have an algorithm that is as simple as their tool checking for particular keys, and that values match for a host of interest.
+2. **Graph-based logic**: A community might warrant using advanced graph parsing to do a similar assessment. An example comes from [Archspec](https://github.com/archspec), where a single architecture (from a flat artifact or the platform, depending on what the tool chooses) can [be placed in a graph](https://github.com/archspec/archspec-json/blob/master/cpu/microarchitectures.json) and used for advanced assessment of compatibility.
+
+The above design encourages simplicity and designation of fewer metadata pieces in the OCI objects. Any complex graph-based query or additional relationships should be represented on the level of the tools that do the compatibility checking. This ensures that changes to these methods or graphs do not require re-creating extensive numbers of artifacts.
+
+## Modifications
+
+This proposal is agnostic to where the JSON object is placed. There are arguably pros and cons to different approaches, and the author (vsoch) leaves it to the expertise of the compatibility working group to decide on a "best" approach.
+
+### Image Manifest
+
+The JSON object could go directly in the image manifest or list to not require any additional artifact query or queries. For runtime tools, this would mean (still) just one call to retrieve all needed metadata to do a check. There would be no additional calls to retrieve plugins. Any compatibility checking plugins (discussed later in this proposal) would already be vetted and installed alongside the runtime tools.
+
+- **Pros**:
+  - Does not require additional query
+  - Cannot get separated from the image manifest or list
+  - Compatibility interest groups can develop artifacts that point TO the image (there is no required approval / permission to develop some niche compatibility definition).
+- **Cons**:
+  - Updating compatibility means also updating the manifest or list
+  - Compatibility interest groups need to advocate to get their metadata added (unlikely / challenging)
+
+### Compatibility Artifact
+
+If desired, a new **Compatibility Artifact** could be added (an OCI artifact) that stores this same JSON object as  a compatibility spec. The artifact would be, akin to other OCI artifacts, stored in a registry separate from an image manifest.
+
+- **Pros**:
+  - Empowers compatibility interest groups (even niche) to create artifacts for images
+- **Cons**:
+  - Can easily be separated from the original manifest / image (would require tooling to check for references)
+  - Requires additional queries
```

Comment:
one more cons. : not all registries support OCI image spec 1.1, has to consider fallback method to check whether compatibility artifact existing

Reply:
Can you clarify this point? Which particular aspect of 1.1 is not supported and what would be the fallback? Arguably if we add some other support for annotations directly in a manifest that would also be a new type that maybe is not supported.

Reply:
for example quay does not support referrer api yet, one issue was open, but no bandwidth to finish it

Reply:
Ah, I understand! So any approach that wants to rely on a separate artifact (using the referers API) would still need a fallback method to check. Likely the simplest thing would be storing the artifact with some known identifier (e.g., :tag -> tag-compat) but that's a bit janky. Worst case, the user / tool is in charge of matching images to artifacts (not linked officially) to check. But the simple approach of storing it in the manifest might work, of course assuming the format of the manifest with said metadata is supported too. It's a hard problem indeed!

Reply:
>  But the simple approach of storing it in the manifest might work

do you mean storing in image manifest?

I have an idea to address the issue if registry does not support referrer api(the fallback method):

1) the image compatibility tool fetches image index first
2) iterate descriptor in the manifests list to pull each manifest
3) check the manifest artifact type, if it's compatibility artifact, then check the subject whether the referred image is the one the tool want to check/validate

through above process, it should be able to find the right compatibility artifact.